src/EnergyPlus/OutputReportTabular.cc

```diff
@@ -721,6 +723,22 @@ void InitializeTabularMonthly(EnergyPlusData &state)
         ort->MonthlyColumns(colNum).duration = 0.0;
     }
 
+    // std::count_if(ort->MonthlyInput.begin(), ort->MonthlyInputCount.end(), [](auto& monthlyInput)
+    // If no weather file run requested, don't bother issuing a warning
+    bool issueWarnings = false;
+    if (state.dataGlobal->DoWeathSim && (state.dataOutRptTab->ErrCount1 > 0)) {
+        ShowWarningError(state, "Processing Monthly Tabular Reports: Variable names not valid for this simulation");
+
+        if (!state.dataGlobal->DisplayExtraWarnings) {
+            ShowContinueError(state, "...use Output:Diagnostics,DisplayExtraWarnings; to show more details on individual variables.");
+        } else {
+            ShowContinueError(state,
+                              "..Variables not valid for this simulation will have \"[Invalid/Undefined]\" in the Units Column of "
+                              "the Table Report.");
+            issueWarnings = true;
+        }
+    }
```

Comment:
We issue the warning if a weather file run period was requested, by checking DoWeathSim and not KindOfSim

Used to not get the message if you have simulation control set to "Run Simulation for Sizing Periods" + "Run Simulation for Weather File Run Period because when this is called, the KindOfSim is DesignDay. This was what caused #8317 actually.



If DisplayExtraWarnings, print the single "Variables not valid for this simulation will have [Invalid/Undefined]" here because it's more logical that printing it after the first guilty var.